File: src/main/java/io/github/dqualizer/dqapi/services/dam/domainstory/ActorService.java
package io.github.dqualizer.dqapi.services.dam.domainstory;

import io.github.dqualizer.dqapi.exceptions.NotFoundException;
import io.github.dqualizer.dqapi.repositories.dam.domainstory.ActorRepository;
import io.github.dqualizer.dqlang.types.dam.domainstory.Actor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ActorService {

    @Autowired
    private ActorRepository repository;

    public List<Actor> readAll() {
        return repository.findAll();
    }

    public Actor readById(String id) {
        return repository.findById(id).orElseThrow(() -> new NotFoundException("Could not find Actor with id: " + id + "."));
    }

    public Actor create(Actor entity) {
        return null; //TODO
    }
}

File: src/main/kotlin/io/github/dqualizer/dqapi/services/dam/domainstory/ActorService.kt
package io.github.dqualizer.dqapi.services.dam.domainstory

import io.github.dqualizer.dqapi.exceptions.NotFoundException
import io.github.dqualizer.dqlang.types.dam.domainstory.Actor
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ActorService {
  @Autowired
  lateinit var repository: io.github.dqualizer.dqapi.repositories.dam.domainstory.ActorRepository

  fun readAll(): List<Actor> {
    return repository.findAll()
  }

  fun readById(id: String): Actor {
    return repository.findById(id).orElseThrow { NotFoundException("Could not find Actor with id: $id.") }
  }

  fun create(entity: Actor?): Actor? {
    TODO()
  }
}
